File: 1880-largest-merge-of-two-strings/largest-merge-of-two-strings.go
func largestMerge(word1 string, word2 string) string {
	var result []byte
	i1, i2 := 0, 0
	for i1 < len(word1) || i2 < len(word2) {
		if i1 < len(word1) && word1[i1:] > word2[i2:] {
			result = append(result, word1[i1])
			i1++
		} else {
			result = append(result, word2[i2])
			i2++
		}
	}
	return string(result)
}

File: 1880-largest-merge-of-two-strings/largest-merge-of-two-strings.rs
impl Solution {
    pub fn largest_merge(word1: String, word2: String) -> String {
        let mut result: Vec<char> = vec!();

        let (mut i, mut j) = (0, 0);

        while i < word1.len() || j < word2.len() {
            if i < word1.len() && word1[i..] > word2[j..] {
                result.push(word1.chars().nth(i).unwrap());
                i += 1;
            } else {
                result.push(word2.chars().nth(j).unwrap());
                j += 1;
            }
        }

        result.into_iter().collect()
    }
}